lib/bms/packet_parsers.cpp

```diff
@@ -17,24 +18,17 @@ inline int16_t int16FromNetworkOrder(const void* const p) {
   return ((uint16_t)(*charPointer)) << 8 | *(charPointer + 1);
 }
 
-int openCircuitSocFromCellVoltage(int cellVoltageMillivolts) {
-  static constexpr int LOOKUP_TABLE_RANGE_MIN_MV = 2700;
-  static constexpr int LOOKUP_TABLE_RANGE_MAX_MV = 4200;
-  static uint8_t LOOKUP_TABLE[31] = {0, 0, 0, 0, 1, 2, 3, 4, 5, 7, 8, 11, 14, 16, 18, 19, 25, 30, 33, 37, 43, 48, 53, 60, 67, 71, 76, 82, 92, 97, 100};
-  static constexpr int LOOKUP_TABLE_SIZE = (sizeof(LOOKUP_TABLE)/sizeof(*LOOKUP_TABLE));
-  static constexpr int RANGE = LOOKUP_TABLE_RANGE_MAX_MV - LOOKUP_TABLE_RANGE_MIN_MV;
-  // (RANGE - 1) upper limit effectively clamps the leftIndex below to (LOOKUP_TABLE_SIZE - 2)
-  cellVoltageMillivolts = clamp(cellVoltageMillivolts - LOOKUP_TABLE_RANGE_MIN_MV, 0, RANGE - 1);
-  float floatIndex = float(cellVoltageMillivolts) * (LOOKUP_TABLE_SIZE - 1) / RANGE;
-  const int leftIndex = int(floatIndex);
-  const float fractional = floatIndex - leftIndex;
-  const int rightIndex = leftIndex + 1;
-  const int leftValue = LOOKUP_TABLE[leftIndex];
-  const int rightValue = LOOKUP_TABLE[rightIndex];
-  return leftValue + round((rightValue - leftValue) * fractional);
+}  // namespace
+
+float BmsRelay::consumedMahPct() {
+  return std::max(((float(-getChargeStateMah()) / float(mah_max_)) * 100.0), 0.0);
+}
+
+int BmsRelay::socFromMahUsed() {
+  float pct = 100 - consumedMahPct();
+  return round(0.8 * pct + 0.00202 * pct * pct); // Magic formula to go from Ah % to Wh %
```

Comment:
Appreciate the effort but I don't think this is worth it. SOC based on Ah and Wh converge close to 100% and 0% SOC and only diverge by ~3% at the highest, see https://docs.google.com/spreadsheets/d/1WluzeQnXibmWL1CxRknn-5j9r9QVfrluIOjotYvx3uU/edit#gid=0, mAhPercent and mWAhPercent columns.

Reply:
I agree its not _required_, however I personally tried both and the tracking definitely feels smoother and more natural with this in place.  Multiplication operations are cheap, I don't think there is harm in having this here.  Whats the actual "cost" in keeping this here?